### PR TITLE
Add FAQ for Scala 3 macros: summon all from tuple

### PR DESCRIPTION
**Pull request by nicolasstucki**
Also fix broken link to ersed-terms documentation.

### CODE REVIEW

File: _overviews/scala3-macros/faq.md [hunk 1]
```diff
@@ -23,3 +23,34 @@ x match
...
+Depending on you use case the way you will summon them will vary.
+The code you will need will depend on the kind of output you want (`Expr[Tuple]`, `List[Expr[Any]]`, or something else) and how you need errors to be reported.
+Here are two examples that should give you the basic skeleton for your variant of this code.
+
+```scala
+  def summonAllInList[T](using Type[T])(using Quotes): List[Expr[Any]] = {
+    Type.of[T] match
+      case '[ head *: tail ] =>
+        Expr.summon[head] match
+          case Some(headExpr) => headExpr :: Expr.summonAll[tail]
```
Comment (bishabosha) [inline _overviews/scala3-macros/faq.md:39]: is `Expr.summonAll` meant to be `summonAllInList`?

Comment (nicolasstucki) [reply _overviews/scala3-macros/faq.md:39]: My bad. It it supposed to be a recursive call to `summonAllInList`

Comment (nicolasstucki) [reply _overviews/scala3-macros/faq.md]: Fixed here and in the second example

File: _overviews/scala3-macros/faq.md [hunk 2]
```diff
@@ -23,3 +23,52 @@ x match
   case '{ $x: t } =>
     // `x: Expr[X & t]` where `t` is the precise type of `x`
 ```
+
+## How do I summon all types of a tuple type?
+If I have a type `(T1, T2, ...)` how do I generate the term for `(summon[T1], summon[T2], ...)` or get the individual expressions with the summoned values?
+
+Depending on you use case the way you will summon them will vary.
```
Comment (b-studios) [inline _overviews/scala3-macros/faq.md:30]:
```suggestion
Depending on your use case the way you will summon them will vary.
```

File: _overviews/scala3-macros/faq.md [hunk 3]
```diff
@@ -23,3 +23,52 @@ x match
...
+The code you will need will depend on the kind of output you want (`Expr[Tuple]`, `List[Expr[Any]]`, or something else) and how you need errors to be reported.
```
Comment (b-studios) [inline _overviews/scala3-macros/faq.md:31]:
```suggestion
In particular, the code you will need depends on the kind of output you want (`Expr[Tuple]`, `List[Expr[Any]]`, or something else) and how you need errors to be reported.
```

File: _overviews/scala3-macros/faq.md [hunk 4]
```diff
@@ -23,3 +23,52 @@ x match
...
+Here are two examples that should give you the basic skeleton for your variant of this code.
```
Comment (b-studios) [inline _overviews/scala3-macros/faq.md:32]:
```suggestion
Here are two examples that should give you the basic skeleton for two different variant of this code.
```

File: _overviews/scala3-macros/faq.md [hunk 5]
```diff
@@ -23,3 +23,52 @@ x match
...
+      case '[ head *: tail ] =>
+        for headExpr <- Expr.summon[head]
+            tailExpr <- summonAll[tail]
+        yield '{ headExpr *: tailExpr }
+      case '[ EmptyTuple ] => Some('{ EmptyTuple })
+      case _ => None
+  }
+```
+
+## How do I summon an expression for statically unknown type
```
Comment (b-studios) [inline _overviews/scala3-macros/faq.md:58]:
```suggestion
## How do I summon an expression for statically unknown types?
```

File: _overviews/scala3-macros/faq.md [hunk 6]
```diff
@@ -23,3 +23,52 @@ x match
...
+
+If you have a `TypeRepr` use
```
Comment (b-studios) [inline _overviews/scala3-macros/faq.md:60]:
```suggestion
You can summon an expression from either a `TypeRepr` or a `Type` as shown below.

If you have a `TypeRepr` use:
```

File: _overviews/scala3-macros/faq.md [hunk 7]
```diff
@@ -23,3 +23,52 @@ x match
...
+```scala
+val tpe: TypeRepr = ...
+Implicits.search(tpe) match
+  case result: ImplicitSearchSuccess => result.tree
+  case _ =>
+```
+
+If there you have a `Type[_]` use
```
Comment (b-studios) [inline _overviews/scala3-macros/faq.md:68]:
```suggestion
Instead, if you have a `Type[_]` use:
```

File: _overviews/scala3-macros/faq.md [hunk 8]
```diff
@@ -23,3 +23,52 @@ x match
...
+```scala
+val tpe: Type[_] = ...
+tpe match
+  // Given the name `a` to the type in tpe and provide a given `Type[a]`
```
Comment (b-studios) [inline _overviews/scala3-macros/faq.md:72]: I don't fully understand this comment here. We do assign a name to our type that we want to search an implicit for, right? What does the latter part of the comment mean?

Comment (nicolasstucki) [reply _overviews/scala3-macros/faq.md:74]: Not sure how to explain it in other words. Technically with the user should assume is that 
```scala
tpe match
  case '[a] => 
    Expr.summon[a]
```

is equivalent to 

```scala
tpe match
  case tpe => 
    type a
    given Type[A] = tpe.asInstanceOf[Type[A]]
    Expr.summon[a]
```

Comment (nicolasstucki) [reply _overviews/scala3-macros/faq.md:74]: This implies that we can use a given `Type[a]` on the RHS of the case. In particular for `Expr.summon[a](using given$Type_a)`.

Comment (b-studios) [reply _overviews/scala3-macros/faq.md:74]: Ok, I see. Maybe 
```
// (1) Use `a` as the name of the unknown type and (2) bring a given `Type[a]` into scope
```
?

Not sure whether this is any better.

**Pull request merged**